### PR TITLE
Resolved Implement OAM Services for version 2.0.0

### LINKED ISSUE #228: Implement OAM Services for version 2.0.0
### **Changes to the existing OAM services:**

> In all the services/<oamservices> modules , modify the import statements to utilize the JSONDriver from the npm package
> var fileOperation = require('../applicationPattern/databaseDriver/JSONDriver');

**TcpClient :**
- [ ] Remove support for the getTcpClientRemoteIpv4Address , putTcpClientRemoteIpv4Address. Remove the functions from the controllers/TcpClient.js and service/TcpClientService.js
- [ ] Include support for the APIs getTcpClientRemoteAddress , putTcpClientRemoteAddress. Copy the functions from the generated server stub , paste it to the controllers/TcpClient.js and service/TcpClientService.js. Implement the service
- [ ] Include support for the APIs getTcpClientRemoteProtocol , putTcpClientRemoteProtocol . Copy the functions from the generated server stub , paste it to the controllers/TcpClient.js and service/TcpClientService.js. Implement the service

**HttpClient :**
- [ ] Include support for the API putHttpClientApplicationName . Copy the functions from the generated server stub , paste it to the controllers/HttpClient.js and service/HttpClientService.js. Implement the service

**TCPServer :**
- [ ] Remove support for the getTcpServerLocalIpv4Address, putTcpServerLocalIpv4Address. Remove the functions from the controllers/TcpServer.js and service/TcpServerService.js
- [ ] Include support for the APIs getTcpServerLocalAddress, putTcpServerLocalAddress. Copy the functions from the generated server stub , paste it to the controllers/TcpServer.js and service/TcpServerService.js. Implement the service
- [ ] Include support for the APIs getTcpServerLocalProtocol, putTcpServerLocalProtocol . Copy the functions from the generated server stub , paste it to the controllers/TcpServer.js and service/TcpServerService.js. Implement the service
- [ ] Include support for the APIs getTcpServerDescription, putTcpServerDescription. Copy the functions from the generated server stub , paste it to the controllers/TcpServer.js and service/TcpServerService.js. Implement the service

### **To include new functionalities of version 2.0.0 :**
**GenericResponseProfile :**
Support Version 2.0.0 changes for GenericResponseProfile
1. Generate server stub for the version 2.0.0 specification 
2. Copy the files controllers/GenericResponseProfile.js and service/GenericResponseProfileService.js from the generated stub to the source code.
3. If the api/openapi.yaml is not updated as per the published version 2.0.0 , then copy the api/openapi.yaml file from the generated stub to the source code
Following tasks needs to be completed to support GenericResponseProfile in version 2.0.0
Implement the logic in the controllers and services for the following services , 
- [ ] getGenericResponseProfileOperationName
- [ ] getGenericResponseProfileFieldName
- [ ] getGenericResponseProfileDescription
- [ ] getGenericResponseProfileDatatype
- [ ] getGenericResponseProfileValue
- [ ] putGenericResponseProfileValue

**ActionProfile :**
Support Version 2.0.0 changes for ActionProfile
1. Generate server stub for the version 2.0.0 specification 
2. Copy the files controllers/ActionProfileProfile.js and service/ActionProfileService.js from the generated stub to the source code.
3. If the api/openapi.yaml is not updated as per the published version 2.0.0 , then copy the api/openapi.yaml file from the generated stub to the source code
Following tasks needs to be completed to support ActionProfileProfile in version 2.0.0
Implement the logic in the controllers and services for the following services ,
- [ ] getActionProfileOperationName
- [ ] getActionProfileLabel
- [ ] getActionProfileInputValueListt
- [ ] getActionProfileDisplayInNewBrowserWindow
- [ ] getActionProfileConsequentOperationReference
- [ ] putActionProfileConsequentOperationReference

**Core :**
Include support for the API getProfileInstance. Copy the functions from the generated server stub , paste it to the controllers/TcpClient.js and service/TcpClientService.js. Implement the service.

### PR DESCRIPTION
**Pull request by ManasaBM1**
Fixes #228

### CODE REVIEW

**Comment by ManasaBM1** [comment]
Reviewer is @DanaSunal

File: server/service/GenericResponseProfileService.js [hunk 1]
```diff
@@ -0,0 +1,133 @@
+'use strict';
+var fileOperation = require('../node_modules/onf-core-model-ap/applicationPattern/databaseDriver/JSONDriver');
```
Comment (DanaSunal) [inline server/service/GenericResponseProfileService.js:2]: This path does not seem correct.

File: server/controllers/OperationServer.js [hunk 2]
```diff
@@ -59,7 +59,7 @@ module.exports.putOperationServerLifeCycleState = async function putOperationSer
 
 module.exports.putOperationServerOperationKey = async function putOperationServerOperationKey(req, res, next, body, uuid) {
   let responseCode = responseCodeEnum.code.NO_CONTENT;
-  await OperationServer.putOperationServerOperationKey(req.url, body)
+  await OperationServer.putOperationServerOperationKey(req.url, body, uuid)
```
Comment (DanaSunal) [inline server/controllers/OperationServer.js:62]: Service method does not use uuid.

File: server/service/TcpServerService.js [hunk 3]
```diff
@@ -79,13 +92,124 @@ exports.putTcpServerLocalIpv4Address = function (url, body) {
...
+    response['application/json'] = {
+      "tcp-server-interface-1-0:description": "tcp-server-interface-1-0:" + value
+    };
+    if (Object.keys(response).length > 0) {
+      resolve(response[Object.keys(response)[0]]);
+    } else {
+      resolve();
+    }
+  });
+}
```
Comment (DanaSunal) [inline server/service/TcpServerService.js:215]: Please remove unnecessary blank lines in whole file.

**Review by DanaSunal — CHANGES_REQUESTED** [review]

File: server/service/GenericResponseProfileService.js (hunk 1 above)
Comment (ManasaBM1) [reply server/service/GenericResponseProfileService.js]: done

File: server/controllers/OperationServer.js (hunk 2 above)
Comment (ManasaBM1) [reply server/controllers/OperationServer.js]: done

File: server/service/TcpServerService.js (hunk 3 above)
Comment (ManasaBM1) [reply server/service/TcpServerService.js:192]: ok done

**Review by DanaSunal — APPROVED** [review]

**Pull request merged**